fmriprep/config.py

```diff
@@ -472,18 +472,24 @@ def init(cls):
             _db_path.mkdir(exist_ok=True, parents=True)
 
             # Recommended after PyBIDS 12.1
+            ignore_patterns = [
+                'code',
+                'stimuli',
+                'sourcedata',
+                'models',
+                re.compile(r'^\.'),
+                re.compile(r'sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/(beh|dwi|eeg|ieeg|meg|perf)'),
+            ]
+            if cls.participant_label and cls.bids_database_dir is None:
+                # Ignore any subjects who aren't the requested ones.
+                # This is only done if the database is written out to a run-specific folder.
+                ignore_patterns.append(
+                    re.compile(r'sub-(?!' + '|'.join(cls.participant_label) + r')\w+')
```

Comment:
Unfortunately, this doesn't ignore `sub-013` if `sub-01` is selected. That's probably okay, as it's an unlikely case and all that happens is indexing more than expected, not less. I'm not sure how to fix up this regex, though. Negative lookaheads are tricky.

Reply:
I don't think I want to fix it in this PR, because I honestly have no clue how to do so, but I want to update the comments to make it clear that this is an issue.

Reply:
Oh, I think I got it:

```suggestion
                    re.compile(r'sub-(?!(' + '|'.join(cls.participant_label) + r')\b)')
```

Reply:
Thanks!